crates/oxc_ast/src/generated/ast_builder.rs

```diff
@@ -3489,16 +5470,36 @@ impl<'a> AstBuilder<'a> {
         Box::new_in(self.catch_parameter(span, pattern), self.allocator)
     }
 
+    /// Builds a [`DebuggerStatement`]
+    ///
+    /// If you want the built node to be allocated in the memory arena, use [`AstBuilder::alloc_debugger_statement`] instead.
+    ///
+    /// ## Parameters
+    /// - span: The [`Span`] covering this node
     #[inline]
     pub fn debugger_statement(self, span: Span) -> DebuggerStatement {
         DebuggerStatement { span }
     }
 
+    /// Builds a [`DebuggerStatement`] and stores it in the memory arena.
+    ///
+    /// Returns a [`Box`] containing the newly-allocated node. If you want a stack-allocated node, use [`AstBuilder::debugger_statement`] instead.
+    ///
+    /// ## Parameters
+    /// - span: The [`Span`] covering this node
     #[inline]
     pub fn alloc_debugger_statement(self, span: Span) -> Box<'a, DebuggerStatement> {
         Box::new_in(self.debugger_statement(span), self.allocator)
     }
 
+    /// Builds a [`BindingPattern`]
+    ///
+    /// If you want the built node to be allocated in the memory arena, use [`AstBuilder::alloc_binding_pattern`] instead.
+    ///
+    /// ## Parameters
+    /// - kind
+    /// - type_annotation
+    /// - optional
```

Comment:
It would be super cool if we add an example here to let the user know what is the ideal usage(especially when some optional arguments are None).

e.g.
```rust
let id = builder.binding_pattern(
    binding_kind,
    Option::<TSTypeAnnotation>::None,
    false
);
```

The compiler tells the user to do this which is less than ideal.

```rust
let id = builder.binding_pattern::<Option<TSTypeAnnotation>>(binding_kind, None, false);
```

Reply:
I just want to add a lot of issues with lots of booleans and optional types would go away as we move toward traced constructors and template macros.
Let me know if you have any suggestions to help with the interface.

Reply:
I agree with the examples. I think this could be tackled in a following PR, so I'm resolving this comment.